connections/ethereum/connection_test.go

```diff
@@ -98,11 +99,52 @@ func TestConnection_SafeEstimateGasMax(t *testing.T) {
 	}
 }
 
+func TestConnection_SafeEstimateGasMin(t *testing.T) {
+	minPrice := big.NewInt(1)
+	// When gasMultipler is zero, the gasPrice is zero if the effect of the minPrice is removed.
+	gasMultipler := big.NewFloat(0)
+	conn := NewConnection(TestEndpoint, false, AliceKp, log15.Root(), GasLimit, MaxGasPrice, minPrice, gasMultipler, "", "")
+	err := conn.Connect()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+
+	price, err := conn.SafeEstimateGas(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if price.Cmp(minPrice) != 0 {
+		t.Fatalf("Gas price should equal min. Suggested: %s Min: %s", price.String(), minPrice.String())
+	}
+}
+
+func TestConnection_SafeEstimateGasSameMin(t *testing.T) {
+	// When GasMultipler is set to 1, the gas price is set to 2000000000, so the minPrice is set to the same price.
+	minPrice := MaxGasPrice
```

Comment:
This test looks good, but could you make the `MaxGasPrice` and `minPrice` different? I know the logic you changed in the estimate gas function isn't very complex I just think it would then be much clearer as to where in the `SafeEstimateGas` method the gas price was actually returned (https://github.com/ChainSafe/ChainBridge/pull/689/files#diff-349cc0f2c50c806250ed335c780d7a6dd9b7ae1cd01fed7e8cfacfa16bba81f6L160-R169). With `minPrice` and  `maxPrice` being the same it just seems much less clear. You could just increase the `maxGasPrice` passed in to `NewConnection` to anything greater than the `minPrice`. Otherwise I think everything is good though.

Reply:
Thank you for your review.
I fixed it https://github.com/ChainSafe/ChainBridge/pull/689/commits/6a6ceeb54e3fc94058a4b77a7ae75ac19c5be2a4.